test/e2e/framework/util.go

```diff
@@ -5100,3 +5103,24 @@ func DsFromManifest(url string) (*extensions.DaemonSet, error) {
 	}
 	return &controller, nil
 }
+
+// WaitForServerPreferredNamespacedResources waits until server preferred namespaced resources could be successfully discovered.
+func WaitForServerPreferredNamespacedResources(d discovery.DiscoveryInterface, timeout time.Duration) ([]*metav1.APIResourceList, error) {
```

Comment:
Make it private and move the TODO here. We don't want this temporary fix be invoked elsewhere.

Reply:
Will do.

Reply:
Done.